pass/ProteusPass.cpp

```diff
@@ -133,27 +132,32 @@ class ProteusJitPassImpl {
     // Host compilation
     // ================
 
+    instrumentRegisterLinkedBinary(M);
+    instrumentRegisterFatBinary(M);
+    instrumentRegisterFatBinaryEnd(M);
+    instrumentRegisterVar(M);
```

Comment:
This needs to be guarded for code that contains device stubs.

Reply:
I checked also the function `hasDeviceLaunchKernelCalls` and is not sufficient. You need to instrument when __hip/cuda_module_ctor is being called.